### ISSUE TITLE
[tune] analysis should not require CSVLogger, JsonLogger

### ISSUE THREAD

**Issue by arunesh-mittal** [ray-project/ray#5522]
### System information

- **OS Platform and Distribution**:

```
NAME="Amazon Linux AMI"
VERSION="2018.03"
ID="amzn"
ID_LIKE="rhel fedora"
VERSION_ID="2018.03"
PRETTY_NAME="Amazon Linux AMI 2018.03"
ANSI_COLOR="0;33"
CPE_NAME="cpe:/o:amazon:linux:2018.03:ga"
HOME_URL="http://aws.amazon.com/amazon-linux-ami/"
```

- **Ray installed from (source or binary)**:
Source 

- **Ray version**:
0.8.0.dev3

- **Python version**:
Python 3.6.5 :: Anaconda, Inc.

- **Exact command to reproduce**:
https://github.com/ray-project/ray/blob/239c177fe8e10c1ca91657bd488132a8d7c57712/python/ray/experimental/sgd/examples/train_example.py

and setting loggers as:
`loggers = [TFLogger]`

### Describe the problem
`analysis.get_best_config(metric="validation_loss", mode="min")` requires both csv and json logger `loggers = [CSVLogger, JsonLogger]` just `TFLogger` raises exception at:

https://github.com/ray-project/ray/blob/3218ee389a3ce1a001826da2efe55c9f0d1508d5/python/ray/tune/analysis/experiment_analysis.py#L67

**Comment by richardliaw** [comment]
Thanks for raising this; I think the correct thing to do here is to have tune keep track of all of the results in memory. For now, this shouldn't be a blocker for you right?

**Comment by richardliaw** [comment]
Addressed in #5962.
